File: app/src/main/java/com/skysoftsolution/in/skill_improvement/padView/DashBoardViewModel.java
package com.skysoftsolution.in.skill_improvement.padView;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.skysoftsolution.in.paymentintegration.dashboard.entity.ModuleForUse;

import java.util.ArrayList;
import java.util.List;

public class DashBoardViewModel extends ViewModel {
    private MutableLiveData<List<ModuleForUse>> userList;

    public DashBoardViewModel() {
        userList = new MutableLiveData<>();
        userList.setValue(new ArrayList<>());
    }

    public LiveData<List<ModuleForUse>> getUserList() {
        return userList;
    }

    public void addModule(ModuleForUse module) {
        List<ModuleForUse> currentList = userList.getValue();
        if (currentList != null) {
            currentList.add(module);
            userList.setValue(currentList);
        }
    }
}

File: app/src/main/java/com/skysoftsolution/in/skill_improvement/dashBoard/viewModel/DashBoardViewModel.kt
package com.skysoftsolution.`in`.paymentintegration.dashboard.viewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.skysoftsolution.`in`.paymentintegration.dashboard.entity.DashBoardModule
import com.skysoftsolution.`in`.paymentintegration.dashboard.entity.ModuleForUse

class DashBoardViewModel : ViewModel() {
    private val _userList = MutableLiveData<DashBoardModule>()
    val userList: LiveData<DashBoardModule> get() = _userList

    init {
        // Initialize with an empty list of users
        _userList.value = DashBoardModule(emptyList())
    }

    fun addModule(user: ModuleForUse) {
        val currentList = _userList.value?.userList ?: emptyList()
        val updatedList = currentList.toMutableList().apply { add(user) }
        _userList.value = DashBoardModule(updatedList)
    }


}